src/controllers/products.js

```diff
@@ -16,3 +16,32 @@ exports.createProduct = async (req, res, next) => {
   const product = await Product.create(req.body);
   return res.status(201).json({ success: true, data: product });
 };
+exports.updateProduct = async (req, res) => {
+  // check for validation errors first
+  const validationErrors = validationResult(req);
+  if (!validationErrors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ success: false, errors: validationErrors.array() });
+  }
+  const product = await Product.findByIdAndUpdate(
```

Comment:
Before updating the product, you need to check if the person trying to update (`req.user`), is the owner of the product.